src/retry_policy/exponential_backoff.rs

```diff
@@ -95,7 +95,11 @@ where
                 debug!(
                     message = "retrying after error",
                     ?error,
-                    backoff_ms = %interval.as_millis()
+                    backoff_ms = %interval.as_millis(),
+                    remaining_attempts = match self.intervals.size_hint() {
+                        (_lower_bound, Some(upper_bound)) => upper_bound,
+                        (lower_bound, None) => lower_bound
```

Comment:
I guess it's not exactly indicative. I thought about having "unknown" if there's no upper bound. In practice this will never matter because the only possible iterator will always have an upper bound (`.take(max_retries.unwrap_or(usize::MAX))`)